frontend/src/model/beta/reviews/ApprovalsDisplay.tsx

```diff
@@ -0,0 +1,50 @@
+import { Done } from '@mui/icons-material'
+import { Stack, Tooltip, Typography } from '@mui/material'
+import { useGetModelRoles } from 'actions/model'
+import { useMemo } from 'react'
+import Loading from 'src/common/Loading'
+import MessageAlert from 'src/MessageAlert'
+import { ReviewResponseWithRole } from 'types/interfaces'
+import { plural } from 'utils/stringUtils'
+
+interface ApprovalsDisplayProps {
+  modelId: string
+  acceptedReviewResponses: ReviewResponseWithRole[]
+}
+
+export default function ApprovalsDisplay({ modelId, acceptedReviewResponses }: ApprovalsDisplayProps) {
+  const { modelRoles, isModelRolesLoading, isModelRolesError } = useGetModelRoles(modelId)
+
+  const roleApprovals = useMemo(() => {
```

Comment:
Looking back at my earlier comment I wasn't very clear and was misleading in places, so my apologies. We should separate concerns with memos as much as possible.

- `staticRoles` should be defined outside of the component (at the top of the file) as it will never change.
- `dynamicRoles` should have its own memo
- The remaining logic can be simplified using `.reduce()`

The 2nd and 3rd points are covered by the code below:
```
  const dynamicRoles = useMemo(() => modelRoles.filter((role) => !staticRoles.includes(role.id)), [modelRoles])

  const roleApprovals = useMemo(
    () =>
      dynamicRoles.reduce<EmotionJSX.Element[]>((approvals, dynamicRole) => {
        const hasRoleApproved = !!acceptedReviewResponses.find(
          (acceptedResponse) => acceptedResponse.role === dynamicRole.id,
        )
        if (hasRoleApproved) {
          approvals.push(
            <Tooltip title={`${plural(acceptedReviewResponses.length, 'approval')}`} key={dynamicRole.id}>
              <Stack direction='row'>
                <Done color='success' fontSize='small' />
                <Typography variant='caption'>{`Approved by ${dynamicRole.name}`}</Typography>
              </Stack>
            </Tooltip>,
          )
        }
        return approvals
      }, []),
    [acceptedReviewResponses, dynamicRoles],
  )
```